zebra-scan/src/service/scan_task/commands.rs

```diff
@@ -43,10 +40,11 @@ pub enum ScanTaskCommand {
     // TODO: Implement this command (#8206)
     SubscribeResults {
         /// Sender for results
-        result_sender: mpsc::Sender<Arc<Transaction>>,
+        // TODO: Update type to return full `WalletTx`
```

Comment:
I think it's okay to make an RPC call from the gRPC method to get the full raw transaction for both the cached and new results from the `scan` RPC (as the 'verbose' response).

It may be be okay to return different types for cached results and newly scanned results, so it doesn't have to cache the full `WalletTx` but returns it for newly scanned results when it's there, but it may also be better to return always the full raw transaction (as the `verbose` response) so it'll be the same for detection keys later.